source/MaterialXRenderGlsl/GlslProgram.cpp

```diff
@@ -371,11 +380,21 @@ void GlslProgram::bindPartition(MeshPartitionPtr partition)
         throw ExceptionShaderRenderError(errorType, errors);
     }
 
-    MeshIndexBuffer& indexData = partition->getIndices();
-    _indexBufferSize = indexData.size();
-    glGenBuffers(1, &_indexBuffer);
-    glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, _indexBuffer);
-    glBufferData(GL_ELEMENT_ARRAY_BUFFER, _indexBufferSize * sizeof(uint32_t), &indexData[0], GL_STATIC_DRAW);
+    const std::string identifier = meshName + partition->getIdentifier();
```

Comment:
Bind the index buffer if it's data is already bound, otherwise, bind the buffer and it's respective data to it.